src/sgx/signing/signing_public.h

```diff
@@ -26,7 +33,29 @@ typedef struct ed25519_context {
    Note: The user must release the enclave by calling release_ed25519_context()
          after they are done using it.
 */
-sgx_status_t init_ed25519(const char* enclave_file, ed25519_context_t* pctxt);
+sgx_status_t init_ed25519(const char* enclave_file,
+                          uint32_t lockout_period,
+                          uint32_t lockout_multiplier,
+                          uint32_t lockout_max_depth,
+                          ed25519_context_t* pctxt);
+
+/* This function returns the sealed data (private key and associated
+   informatio). The sealed data can be used to reinit the enclave using
+   init_ed25519_from_data().
+*/
+sgx_status_t get_ed25519_data(ed25519_context_t* pctxt,
+                              uint32_t* datalen,
+                              uint8_t* data);
+
+/* This function reinitializes the enclave using sealed data.
+ */
+sgx_status_t init_ed25519_from_data(ed25519_context_t* pctxt,
+                                    uint32_t datalen,
+                                    uint8_t* data,
+                                    uint32_t update_lockout_params,
+                                    uint32_t lockout_period,
```

Comment:
lockout period shouldn't be re-initialize-able, should be part of initial setup

Reply:
👍 
I think APIs will be updated based on final design of the service. I am leaving them as is for now, as the signatures are still in flux.

Per your suggestion, I've added code for rollback protection. That'll be useful to protect against reinit.